islandora.module

```diff
@@ -1226,7 +1226,13 @@ function islandora_object_load($object_id) {
       return $tuque->repository->getObject(urldecode($object_id));
     }
     catch (Exception $e) {
-      if ($e->getCode() == '404') {
+      // Check to see if tuque failure is due to invalid PID.
+      module_load_include('inc', 'islandora', 'includes/utilities');
+      if (!islandora_is_valid_pid($object_id)) {
+        watchdog('islandora', 'Failed to load invalid PID: @pid', array('@pid' => $object_id), WATCHDOG_ERROR);
```

Comment:
@bryjbrown looks good! As a fanatical system admin (a hat a have to wear sometimes) I wonder if everybody needs this watchdog message or could you/others be fine with simply avoiding it. Mostly because this could potentially become a long list in our log?. More like a question than a big concern really and I already, upfront, approve your pull request, good work!. I just open this question because 404 can also be logged directly on a Drupal level if someone wants to/needs that

Reply:
But I also understand the watchdog (monolog sorry). Its a way to distinguish/refine the reason why we throw the 404

Reply:
@DiegoPino True enough, all the requests for malformed PIDs result in 404s, and as a result get a "Page not found" notice in the log, but they are generic 'warnings' whereas this is a very specific 'error' which may possibly be useful for log filtering.

I don't disagree with you though, just pointing out my rationale for keeping it in. I don't have a strong opinion either way, and mainly added it to help me figure out where to add the check. Any other @Islandora/7-x-1-x-committers have a preference?

Reply:
My 2 cents: I don't think we need to log this, its probably okay to just have the "page not found" notice in the log.

Reply:
@jonathangreen @DiegoPino Fair enough, I'll remove it from this PR and the upcoming release branch PR. :+1: